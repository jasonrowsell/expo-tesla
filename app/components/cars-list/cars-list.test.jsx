import React from 'react';
import { render } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import CarsList from './cars-list';

describe('<CarsList />', () => {
  test('has 1 child', () => {
    const tree = renderer.create(<CarsList title="title" tagline="tagline" source="img" />).toJSON();
    expect(tree.children.length).toBe(1);
  });

  test('renders correctly', () => {
    const tree = render(<CarsList />);
    expect(tree).toMatchSnapshot();
  });
});
