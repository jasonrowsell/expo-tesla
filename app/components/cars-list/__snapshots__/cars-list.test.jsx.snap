// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarsList /> renders correctly 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <RCTScrollView
    alwaysBounceVertical={true}
    data={
      Array [
        Object {
          "image": 1,
          "tagline": "1% Company Car Tax.",
          "taglineCTA": "Learn More",
          "title": "Model 3",
        },
        Object {
          "image": 1,
          "tagline": "Order Online for",
          "taglineCTA": "Touchless Delivery",
          "title": "Model S",
        },
        Object {
          "image": 1,
          "tagline": "Order Online for",
          "taglineCTA": "Touchless Delivery",
          "title": "Model X",
        },
        Object {
          "image": 1,
          "tagline": "Order Online for",
          "taglineCTA": "Touchless Delivery",
          "title": "Model Y",
        },
      ]
    }
    decelerationRate="fast"
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={16}
    scrollY={0}
    showsVerticalScrollIndicator={false}
    snapToAlignment="start"
    snapToInterval={1334}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flex": 1,
              "height": 1334,
              "width": "100%",
            }
          }
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Object {
                "height": "100%",
                "position": "absolute",
                "resizeMode": "cover",
                "width": "100%",
              }
            }
          >
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": "100%",
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "marginTop": "30%",
                "opacity": 1,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "display": "flex",
                  "fontSize": 40,
                  "fontWeight": "500",
                  "paddingBottom": 5,
                }
              }
            >
              Model 3
            </Text>
            <Text
              style={
                Object {
                  "color": "#5c5e62",
                  "fontSize": 17,
                }
              }
            >
              1% Company Car Tax.
               
              <Text
                style={
                  Object {
                    "textDecorationLine": "underline",
                  }
                }
              >
                Learn More
              </Text>
            </Text>
          </View>
          <View
            style={
              Object {
                "bottom": 50,
                "opacity": 1,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "padding": 10,
                  "width": "100%",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 20,
                      "height": 40,
                      "justifyContent": "center",
                    },
                    Object {
                      "backgroundColor": "#171A20CC",
                    },
                  ]
                }
                testID="button"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 14,
                        "fontWeight": "500",
                        "textTransform": "uppercase",
                      },
                      Object {
                        "color": "#FFFFFF",
                      },
                    ]
                  }
                >
                  Custom Order
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "padding": 10,
                  "width": "100%",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 20,
                      "height": 40,
                      "justifyContent": "center",
                    },
                    Object {
                      "backgroundColor": "#FFFFFFA6",
                    },
                  ]
                }
                testID="button"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 14,
                        "fontWeight": "500",
                        "textTransform": "uppercase",
                      },
                      Object {
                        "color": "#171A20",
                      },
                    ]
                  }
                >
                  Available Inventory
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flex": 1,
              "height": 1334,
              "width": "100%",
            }
          }
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Object {
                "height": "100%",
                "position": "absolute",
                "resizeMode": "cover",
                "width": "100%",
              }
            }
          >
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": "100%",
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "marginTop": "30%",
                "opacity": -1.333333333333333,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "display": "flex",
                  "fontSize": 40,
                  "fontWeight": "500",
                  "paddingBottom": 5,
                }
              }
            >
              Model S
            </Text>
            <Text
              style={
                Object {
                  "color": "#5c5e62",
                  "fontSize": 17,
                }
              }
            >
              Order Online for
               
              <Text
                style={
                  Object {
                    "textDecorationLine": "underline",
                  }
                }
              >
                Touchless Delivery
              </Text>
            </Text>
          </View>
          <View
            style={
              Object {
                "bottom": 50,
                "opacity": -1.333333333333333,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "padding": 10,
                  "width": "100%",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 20,
                      "height": 40,
                      "justifyContent": "center",
                    },
                    Object {
                      "backgroundColor": "#171A20CC",
                    },
                  ]
                }
                testID="button"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 14,
                        "fontWeight": "500",
                        "textTransform": "uppercase",
                      },
                      Object {
                        "color": "#FFFFFF",
                      },
                    ]
                  }
                >
                  Custom Order
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "padding": 10,
                  "width": "100%",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 20,
                      "height": 40,
                      "justifyContent": "center",
                    },
                    Object {
                      "backgroundColor": "#FFFFFFA6",
                    },
                  ]
                }
                testID="button"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 14,
                        "fontWeight": "500",
                        "textTransform": "uppercase",
                      },
                      Object {
                        "color": "#171A20",
                      },
                    ]
                  }
                >
                  Available Inventory
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flex": 1,
              "height": 1334,
              "width": "100%",
            }
          }
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Object {
                "height": "100%",
                "position": "absolute",
                "resizeMode": "cover",
                "width": "100%",
              }
            }
          >
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": "100%",
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "marginTop": "30%",
                "opacity": -3.666666666666664,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "display": "flex",
                  "fontSize": 40,
                  "fontWeight": "500",
                  "paddingBottom": 5,
                }
              }
            >
              Model X
            </Text>
            <Text
              style={
                Object {
                  "color": "#5c5e62",
                  "fontSize": 17,
                }
              }
            >
              Order Online for
               
              <Text
                style={
                  Object {
                    "textDecorationLine": "underline",
                  }
                }
              >
                Touchless Delivery
              </Text>
            </Text>
          </View>
          <View
            style={
              Object {
                "bottom": 50,
                "opacity": -3.666666666666664,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "padding": 10,
                  "width": "100%",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 20,
                      "height": 40,
                      "justifyContent": "center",
                    },
                    Object {
                      "backgroundColor": "#171A20CC",
                    },
                  ]
                }
                testID="button"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 14,
                        "fontWeight": "500",
                        "textTransform": "uppercase",
                      },
                      Object {
                        "color": "#FFFFFF",
                      },
                    ]
                  }
                >
                  Custom Order
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "padding": 10,
                  "width": "100%",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 20,
                      "height": 40,
                      "justifyContent": "center",
                    },
                    Object {
                      "backgroundColor": "#FFFFFFA6",
                    },
                  ]
                }
                testID="button"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 14,
                        "fontWeight": "500",
                        "textTransform": "uppercase",
                      },
                      Object {
                        "color": "#171A20",
                      },
                    ]
                  }
                >
                  Available Inventory
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flex": 1,
              "height": 1334,
              "width": "100%",
            }
          }
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Object {
                "height": "100%",
                "position": "absolute",
                "resizeMode": "cover",
                "width": "100%",
              }
            }
          >
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": "100%",
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "marginTop": "30%",
                "opacity": -6.0000000000000036,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "display": "flex",
                  "fontSize": 40,
                  "fontWeight": "500",
                  "paddingBottom": 5,
                }
              }
            >
              Model Y
            </Text>
            <Text
              style={
                Object {
                  "color": "#5c5e62",
                  "fontSize": 17,
                }
              }
            >
              Order Online for
               
              <Text
                style={
                  Object {
                    "textDecorationLine": "underline",
                  }
                }
              >
                Touchless Delivery
              </Text>
            </Text>
          </View>
          <View
            style={
              Object {
                "bottom": 50,
                "opacity": -6.0000000000000036,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "padding": 10,
                  "width": "100%",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 20,
                      "height": 40,
                      "justifyContent": "center",
                    },
                    Object {
                      "backgroundColor": "#171A20CC",
                    },
                  ]
                }
                testID="button"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 14,
                        "fontWeight": "500",
                        "textTransform": "uppercase",
                      },
                      Object {
                        "color": "#FFFFFF",
                      },
                    ]
                  }
                >
                  Custom Order
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "padding": 10,
                  "width": "100%",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 20,
                      "height": 40,
                      "justifyContent": "center",
                    },
                    Object {
                      "backgroundColor": "#FFFFFFA6",
                    },
                  ]
                }
                testID="button"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 14,
                        "fontWeight": "500",
                        "textTransform": "uppercase",
                      },
                      Object {
                        "color": "#171A20",
                      },
                    ]
                  }
                >
                  Available Inventory
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
